controller/embedding/util.py

```diff
@@ -1,5 +1,34 @@
-from submodules.model.business_objects import embedding
+from controller.embedding import connector
+from submodules.model import enums
+from submodules.model.business_objects import agreement, embedding, general
+from util import daemon
 
 
 def has_encoder_running(project_id: str) -> bool:
     return embedding.get_first_running_encoder(project_id) is not None
+
+
+def recreate_embedding(
+    project_id: str, embedding_id: str
+) -> None:
+    old_embedding_item = embedding.get(project_id, embedding_id)
+    agreement_item = agreement.get(project_id, old_embedding_item.agreement_id)
+    new_embedding_item = embedding.create(
+        project_id,
+        old_embedding_item.attribute_id,
+        old_embedding_item.name,
+        enums.EmbeddingState.INITIALIZING.value,
```

Comment:
is this correct? maybe i misunderstand the method. I though recreate_embedding can be called from bacially everywhere and it handles the whole recreation thing. If it's just for the upload of new records (so not task queue related) then we should note that somewhere.